NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -256,3 +415,517 @@ def make_sim_shower(corsika, observer=None, detector=None, station_id=None):
+
+        ### Get interpolated values at each grid point, calling the instance of interp2d_fourier
+        ZI =  self.fluence_interpolator(XI, YI)
+
+        # And plot it
+        maxp = np.max(ZI)
+        fig, ax = plt.subplots(figsize=(8, 6))
+        ax.pcolor(XI, YI, ZI, vmax=maxp, vmin=0, cmap=cm.gnuplot2_r)
+        mm = cm.ScalarMappable(cmap=cm.gnuplot2_r)
+        mm.set_array([0.0, maxp])
+        cbar = plt.colorbar(mm, ax=ax)
+        cbar.set_label(r'energy fluence [eV/m^2]', fontsize=14)
+        ax.set_xlabel(r'$\vec{v} \times \vec{B} [m]', fontsize=16)
+        ax.set_ylabel(r'$\vec{v} \times (\vec{v} \times \vec{B})$ [m]', fontsize=16)
+        ax.set_xlim(-radius, radius)
+        ax.set_ylim(-radius, radius)
+        ax.set_aspect('equal')
+        return fig, ax
+
+    def get_interp_efield_value(self, position_on_ground, core):
+        """
+        Accesses the interpolated electric field given the position of the detector on ground. For the interpolation, the pulse will be
+        projected in the shower plane. If the geomagnetic angle is smaller than 15deg, the electric field of the closest observer position is returned.
+
+        Parameters
+        ----------
+        position_on_ground : np.array (3)
+            position of the antenna on ground
+
+        core : np.array (3)
+            position of the core on ground
+
+        Returns
+        -------
+        efield_interp : float
+            interpolated efield value or efield of clostest observer
+
+        trace_start_time : float
+            start time of the trace
+        """
+        logger.debug(f"get interpolated efield for antenna position {position_on_ground} on ground and core position {core}")
+        antenna_position = copy.copy(position_on_ground)
+        #core and antenna need to be in the same z plane
+        antenna_position[2] = core[2]
+        # transform antenna position into shower plane with respect to core position, core position is set to 0,0 in shower plane
+        antenna_pos_vBvvB = self.cs.transform_to_vxB_vxvxB(antenna_position, core=core)
+        logger.debug(f"antenna position in shower plane {antenna_pos_vBvvB}")
+
+        # calculate distance between core position at(0,0) and antenna positions in shower plane
+        dcore_vBvvB = np.linalg.norm(antenna_pos_vBvvB[:-1])
+        # interpolate electric field at antenna position in shower plane which are inside star pattern
+        if dcore_vBvvB > self.star_radius:
+            efield_interp = self.empty_efield
+            logger.debug(f'antenna position with distance {dcore_vBvvB:.2f} to core is outside of star pattern with radius {self.star_radius:.2f} on ground {self.geo_star_radius:.2f}, set efield and fluence to zero')
+        else:
+            if self.efield_interpolator == -1:
+                efield = self.get_closest_observer_efield(antenna_pos_vBvvB)
+                efield_interp = efield
+                trace_start_time = None
+
+            else:
```

Comment:
Is the `const_time_offset` variable necessary if we already use `pulse_centered=True`?

Reply:
I am happy to get rid of it.